src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -325,9 +325,144 @@ module.exports = class LevelPlane extends Array {
       posToRefresh.push(this.redstoneListON[i]);
     }
 
+    //once we're done updating redstoneWire states, check to see if doors should open/close
+    for (let i = 0; i < this.length; ++i) {
+      if (this[i].blockType === "doorIron") {
+        this[i].isPowered = this.powerCheck(this.indexToCoordinates(i));
+        if (this[i].isPowered && !this[i].isOpen) {
+          this.animateDoor(i, true);
+        } else if (!this[i].isPowered && this[i].isOpen) {
+          this.animateDoor(i, false);
+        }
+      }
+
+      if (this[i].blockType.startsWith("piston")) {
+        this[i].isPowered = this.powerCheck(this.indexToCoordinates(i));
+        if (this[i].isPowered) {
+          this.activatePiston(this.indexToCoordinates(i));
+          this.levelModel.controller.updateFowPlane();
+          this.levelModel.controller.updateShadingPlane();
+        } else if (!this[i].isPowered) {
+          this.deactivatePiston(this.indexToCoordinates(i));
```

Comment:
We're calling `this.indexToCoordinates(i)` repeatedly in this added code.  Can we convert once and work with `position` instead of index?

Better yet, let's iterate through the positions so we're not directly dealing with indices until we need to actually look up items in the array.  See my feedback here: https://github.com/code-dot-org/craft/pull/67/files#r137057705.